crev-lib/src/local.rs

```diff
@@ -591,8 +616,8 @@ impl Local {
         let mut already_fetched_ids = HashSet::new();
         let mut already_fetched_urls = HashSet::new();
         let mut db = crate::ProofDB::new();
-        db.import_from_iter(self.proofs_iter()?);
-        db.import_from_iter(proofs_iter_for_path(self.cache_remotes_path()));
+        db.import_from_iter(self.proofs_for_current_url()?);
```

Comment:
Here I'm getting confused.

`fetch_trusted` is supposed to fetch updates from trusted **Ids**. It basically imports all proofs , then iterate over each trusted id, fetch update, import what was cached and so on.

I don't understand why would we need `self.proofs_for_current_url()` here. Seems to me that it switches everything to rely on urls, while before we were using much stronger source of urls - self-reported url, signed by each id. We only switched back to url's reported by other IDs, if a self-reported one was not found.

Reply:
`proofs_for_current_url` is `proofs_iter`. I only renamed it. I don't know why you put it there…

Reply:
`self.proofs_iter` AFAIR is supposed to iterate over all proofs in the local copy of the proof repository for the current user. The url of the user is irrelevant, and if the user has multiple ids sharing the repository they will all be imported.

`proofs_iter` + `cache_remotes_path` is pretty much "load all the proofs that we have locally",

Reply:
No, proofs_iter doesn't iterate all local proofs. It calls `get_proofs_dir_path_opt` which is valid for 1 URL only:

https://github.com/crev-dev/cargo-crev/blob/6ee3dfe78cbffaa4b0981006fba501aa2364601e/crev-lib/src/local.rs#L986-L988

https://github.com/crev-dev/cargo-crev/blob/6ee3dfe78cbffaa4b0981006fba501aa2364601e/crev-lib/src/local.rs#L515-L516

Reply:
Iterating all of them would make sense, so I can change that.

Reply:
Oh... You're right. Anyway, yes, I see no reason to not use any other repos the user might have, even if they switched to a different user or something.

Reply:
As mentioned elsewhere, `proofs_for_current_url` was merely a corrected name for existing behavior that is on master currently.

These two lines were essentially a copy-paste of `load_db` function, so I've replaced them with a call to `load_db()`.